src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+    else:
+        idx = np.argsort(pred_x_vals)
+        x_data = pred_x_vals[idx]
+        y_data = ci_data[idx]
+
+    ci_lower = y_data[:, 0]
+    ci_upper = y_data[:, 1]
+
+    # upper and lower lines of credible interval
+    line_kwargs = copy(visuals.get("line", {}))
+    if line_kwargs is not False:
+        _, _, line_ignore = filter_aes(plot_collection, aes_by_visuals, "line", sample_dims)
+        plot_collection.map(
+            line_xy,
+            "lower_ci_line",
+            x=x_data,
+            y=ci_lower,
+            ignore_aes=line_ignore,
+            **line_kwargs,
+        )
+
+        plot_collection.map(
+            line_xy,
+            "upper_ci_line",
+            x=x_data,
+            y=ci_upper,
+            ignore_aes=line_ignore,
+            **line_kwargs,
+        )
+
+    # data prepration for `fill` visual
+    data = np.stack([x_data, ci_lower, ci_upper], axis=0)
+    da = xr.DataArray(
+        data,
+        dims=["kwarg", "index"],
+        coords={
+            "kwarg": ["x", "y_bottom", "y_top"],
+            "index": np.arange(len(x_data)),
+        },
+        name="ci_bounds",
+    )
+
+    # fill between lines of credible interval
+    fill_kwargs = copy(visuals.get("fill", {}))
+    if fill_kwargs is not False:
+        _, _, fill_ignore = filter_aes(plot_collection, aes_by_visuals, "fill", sample_dims)
+        plot_collection.map(
+            fill_between_y,
+            "fill",
+            data=da,
+            ignore_aes=fill_ignore,
+            color=colors[1],
+            alpha=0.4,
+            **fill_kwargs,
+        )
+
+    # mean line
+    mean_line_kwargs = copy(visuals.get("mean_line", {}))
+    if mean_line_kwargs is not False:
+        _, _, mean_line_ignore = filter_aes(
+            plot_collection, aes_by_visuals, "mean_line", sample_dims
+        )
+        plot_collection.map(
+            line_xy,
+            "mean_line",
+            x=pred_x_vals,
+            y=mean_data.y.values,
+            ignore_aes=mean_line_ignore,
+            **mean_line_kwargs,
+        )
+
+    # scatter plot
+    original_scatter_kwargs = copy(visuals.get("scatter", {}))
+    if original_scatter_kwargs is not False:
+        _, _, scatter_ignore = filter_aes(plot_collection, aes_by_visuals, "scatter", sample_dims)
+        plot_collection.map(
+            scatter_xy,
+            "scatter",
+            x=independent_data[independent_var].values,
+            y=target_data[target_var].values,
+            alpha=0.3,
+            color=contrast_gray_color,
+            width=0,
+            ignore_aes=scatter_ignore,
+            **original_scatter_kwargs,
+        )
+
+    # x-axis label
+    xlabel_kwargs = copy(visuals.get("xlabel", {}))
+
+    if xlabel_kwargs is not False:
+        _, _, xlabel_ignore = filter_aes(plot_collection, aes_by_visuals, "xlabel", sample_dims)
+        plot_collection.map(
+            labelled_x,
+            "xlabel",
+            text="Predictor",
```

Comment:
we can probably try and get a better default (`independent_data` as a dataarray might have a name we could use here). But in either case, this should go into `**xlabel_kwargs` through `setdefault` so users can choose the text if desired. Same for ylabel